fontforgeexe/metricsview.c

```diff
@@ -1542,16 +1542,26 @@ static int MV_KernChanged(GGadget *g, GEvent *e) {
 return( true );
     if ( which>mv->glyphcnt-1 || which==0 )
 return( true );
+
     if ( e->u.control.subtype == et_textchanged ) {
-	unichar_t *end;
-	int val = u_strtol(_GGadgetGetTitle(g),&end,10);
+        char* title = GGadgetGetTitle8(g);
 
-	if ( *end && !(*end=='-' && end[1]=='\0'))
-	    GDrawBeep(NULL);
-	else {
-	    MV_ChangeKerning(mv,which,val, false);
-	    MVRemetric(mv);
-	}
+        int negatives = 0;
+        for (char* p = title; *p != '\0'; p++) {
+            if (*p == '-') negatives++;
+        }
+
+        char* new = str_replace_all(title, "-", "", true); // frees title
```

Comment:
Unfortunately neither the for loop above nor this call is likely "safe". Looking for a hyphen is looking for a byte value of 45. In UTF 8 that byte could be part of a multibyte character. That will throw off your negative calculation, but more importantly *eliminating* that byte, as with `str_replace_all`, could corrupt the whole string (by shifting one or more characters into the multibyte character).

How much does this matter for a field where someone is supposed to just type a number? Hmm -- maybe not much? Probably not much?

Reply:
I don't really think this matters. The worst that can happen is a number turns negative, but junk is already being entered. Only ASCII works in this field; as the saying goes, "garbage in, garbage out."